src/core/mount.c

```diff
@@ -452,7 +452,17 @@ static int mount_add_default_dependencies(Mount *m) {
         mask = m->from_fragment ? UNIT_DEPENDENCY_FILE : UNIT_DEPENDENCY_MOUNTINFO_DEFAULT;
         nofail = m->from_fragment ? fstab_test_yes_no_option(m->parameters_fragment.options, "nofail\0" "fail\0") : false;
 
-        if (mount_is_network(p)) {
+        e = path_startswith(m->where, "/sysroot");
+        if (e && in_initrd()) {
+
+                /* All mounts under /sysroot need to happen later, at initrd-fs.target time. IOW,
+                 * it's not technically part of the basic initrd filesystem itself, and so
+                 * shouldn't inherit the default Before=local-fs.target dependency. */
+
+                after = NULL;
+                before = isempty(e) ? SPECIAL_INITRD_ROOT_FS_TARGET : SPECIAL_INITRD_FS_TARGET;
```

Comment:
isempty() → empty_or_root(), no?

Reply:
It depends on whether `m->where` can have a trailing '/'. According to the code, this path is simplified so it shouldn't have it so at a first glance `empty_or_root()` doesn't look necessary.

Reply:
well, but it's easier to read and more robust if we make clear what is happening here, by writing it down as `empty_or_root()`, no?